File: frontend/src/objects/domain/AfterSalesRequestType.ts
export const AFTER_SALES_REQUEST_TYPE = {
  returnRequest: 'ReturnRequest',
  compensationRequest: 'CompensationRequest',
} as const

export type AfterSalesRequestType =
  (typeof AFTER_SALES_REQUEST_TYPE)[keyof typeof AFTER_SALES_REQUEST_TYPE]

File: backend/src/main/scala/domain/shared/AfterSalesRequestType.scala
package domain.shared

import domain.shared.given

import io.circe.{Decoder, Encoder}

enum AfterSalesRequestType:
  case ReturnRequest, CompensationRequest

object AfterSalesRequestType:
  private val enumLabel = text("AfterSalesRequestType")
  def render(value: AfterSalesRequestType): DisplayText = enumDisplayText(value)
  def parse(value: DisplayText): Option[AfterSalesRequestType] = parseEnumValue(value, AfterSalesRequestType.values)
  given Encoder[AfterSalesRequestType] = enumEncoder
  given Decoder[AfterSalesRequestType] = enumDecoder(AfterSalesRequestType.values, enumLabel)
